dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/imports/bundle/EventImportTest.java

```diff
@@ -0,0 +1,160 @@
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+import java.io.IOException;
+import java.util.stream.Stream;
+import org.hisp.dhis.common.UID;
+import org.hisp.dhis.event.EventStatus;
+import org.hisp.dhis.feedback.ForbiddenException;
+import org.hisp.dhis.feedback.NotFoundException;
+import org.hisp.dhis.program.Event;
+import org.hisp.dhis.tracker.TrackerTest;
+import org.hisp.dhis.tracker.export.event.EventService;
+import org.hisp.dhis.tracker.imports.TrackerImportParams;
+import org.hisp.dhis.tracker.imports.TrackerImportService;
+import org.hisp.dhis.tracker.imports.domain.TrackerObjects;
+import org.hisp.dhis.tracker.imports.report.ImportReport;
+import org.hisp.dhis.user.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.springframework.beans.factory.annotation.Autowired;
+
+class EventImportTest extends TrackerTest {
+  @Autowired private TrackerImportService trackerImportService;
+
+  @Autowired private EventService eventService;
+
+  private User importUser;
+
+  @BeforeAll
+  void setUp() throws IOException {
+    setUpMetadata("tracker/simple_metadata.json");
+
+    importUser = userService.getUser("tTgjgobT1oS");
+    injectSecurityContextUser(importUser);
+  }
+
+  @Test
+  void shouldPopulateCompletedDataWhenCreatingAnEventWithStatusCompleted()
+      throws IOException, ForbiddenException, NotFoundException {
+    TrackerImportParams params = TrackerImportParams.builder().userId(importUser.getUid()).build();
+    TrackerObjects trackerObjects = fromJson("tracker/te_enrollment_event.json");
+    trackerObjects.getEvents().get(0).setStatus(EventStatus.COMPLETED);
+
+    ImportReport importReport = trackerImportService.importTracker(params, trackerObjects);
+
+    assertNoErrors(importReport);
+
+    Event event = eventService.getEvent(UID.of("D9PbzJY8bJO"));
+
+    assertEquals(importUser.getUsername(), event.getCompletedBy());
+    assertNotNull(event.getCompletedDate());
+  }
+
+  @ParameterizedTest
+  @MethodSource("notCompletedStatus")
+  void shouldNotPopulateCompletedDataWhenCreatingAnEventWithNotCompletedStatus(EventStatus status)
+      throws IOException, ForbiddenException, NotFoundException {
+    TrackerImportParams params = TrackerImportParams.builder().userId(importUser.getUid()).build();
+    TrackerObjects trackerObjects = fromJson("tracker/te_enrollment_event.json");
+    trackerObjects.getEvents().get(0).setStatus(status);
+
+    ImportReport importReport = trackerImportService.importTracker(params, trackerObjects);
+
+    assertNoErrors(importReport);
+
+    Event event = eventService.getEvent(UID.of("D9PbzJY8bJO"));
+
+    assertNull(event.getCompletedBy());
+    assertNull(event.getCompletedDate());
+  }
+
+  @Test
+  void shouldDeleteCompletedDataWhenUpdatingAnEventWithStatusActive()
```

Comment:
I think you should merge this with the first test as you should assert that `completed` fields are set before than setting status to active and asserting they are not.